onboard/catalog_actions.go

```diff
@@ -23,19 +28,38 @@ type CatalogRepoActions struct {
 	cataloger      catalog.Cataloger
 	repository     string
 	committer      string
+	logger         logging.Logger
+}
+
+func NewCatalogActions(cataloger catalog.Cataloger, repository string, committer string, logger logging.Logger) RepoActions {
+	return &CatalogRepoActions{cataloger: cataloger, repository: repository, committer: committer, logger: logger}
 }
 
-func NewCatalogActions(cataloger catalog.Cataloger, repository string, committer string) RepoActions {
-	return &CatalogRepoActions{cataloger: cataloger, repository: repository, committer: committer}
+type task struct {
+	f   func() error
+	err *error
+}
+
+func worker(wg *sync.WaitGroup, tasks <-chan *task) {
+	for task := range tasks {
+		*task.err = task.f()
+		wg.Done()
+	}
 }
 
 func (c *CatalogRepoActions) ApplyImport(ctx context.Context, it Iterator, dryRun bool) (*InventoryImportStats, error) {
 	var stats InventoryImportStats
+	var wg sync.WaitGroup
 	batchSize := DefaultWriteBatchSize
 	if c.WriteBatchSize > 0 {
 		batchSize = c.WriteBatchSize
 	}
+	errs := make([]*error, 0)
+	tasksChan := make(chan *task)
```

Comment:
This is a channel with capacity 1, it is limited in how much parallelism it can add.  It should be around `2 * DefaultWorkerCount` which will make starving the workers harder.